merkle/smt/hstar3.go

```diff
@@ -43,7 +43,7 @@ type HashChildrenFn func(left, right []byte) []byte
 //
 // TODO(pavelkalinnikov): Swap in the code, and document it properly.
 type HStar3 struct {
-	upd   []NodeUpdate
+	upd   []Node
```

Comment:
The name `upd` is now not quite as relevant. Maybe `nodes` or similar is better.